components/rate/rate.component.ts

```diff
@@ -220,10 +220,14 @@ export class NzRateComponent implements OnInit, ControlValueAccessor, OnChanges
   }
 
   onItemHover(index: number, isHalf: boolean): void {
-    if (this.nzDisabled || (this.hoverValue === index + 1 && isHalf === this.hasHalf)) {
+    if (this.nzDisabled) {
       return;
     }
 
+    if (this.hoverValue === index + 1 && isHalf === this.hasHalf) {
+      return this.nzOnHoverChange.emit(this.hoverValue);
+    }
+
```

Comment:
I think the logic here would be this, WDYT :)

```ts
  onItemHover(index: number, isHalf: boolean): void {
    if (this.nzDisabled) {
      return;
    }

    if (this.hoverValue !== index + 1 || isHalf !== this.hasHalf) {
      this.hoverValue = index + 1;
      this.hasHalf = isHalf;
      this.updateStarStyle();
    }

    this.nzOnHoverChange.emit(this.hoverValue);
  }

```

Reply:
Hi, I'm not against this proposal quite the contrary :), but if we do that we break the current behavior of the component isn't it ? Maybe we could do a little RFC ?

Reply:
https://stackblitz.com/edit/react-efsuqc?file=demo.tsx

Based on my observations of the behavior of the antd rate component, I think this is a bug fix.

If it is a breaking change, we will need the next major release before we can merge it.

Reply:
@HyperLife1119 after investigation on my side you're right, so i made the change :). Thanks for your time and help

Reply:
Oh, I think you're missing some details:
- When disabled, events are not triggered.
- The event still needs to be triggered even if the rate value has not changed.

I think it's supposed to work like this:

```ts
  onItemHover(index: number, isHalf: boolean): void {
    if (this.nzDisabled) {
      return;
    }

    if (this.hoverValue !== index + 1 || isHalf !== this.hasHalf) {
      this.hoverValue = index + 1;
      this.hasHalf = isHalf;
      this.updateStarStyle();
    }

    this.nzOnHoverChange.emit(this.hoverValue);
  }
```

Reply:
Great job, Nicoss :)